content/en/blog/2024/getting-started-with-otelsql/index.md

```diff
@@ -0,0 +1,428 @@
+		err = rows.Scan(&currentTime)
+		if err != nil {
+			return err
+		}
+	}
+	fmt.Println(currentTime)
+	return nil
+}
+```
+
+This `runSQLQuery` method creates a parent span first (it is an optional step,
+it makes the query spans have a parent, and it looks good on the trace graph),
+then queries the current timestamp from the MySQL database.
+
+After this method, the `client` application finished and exited. And, these are
+all the most important lines of code for understanding the example.
+
+## Use the example as a playground
+
+After understanding the example, we can use it as a playground, making it a bit
+complicated to see how it will be used in a real-world-like case.
+
+Use the following codes to replace the `runSQLQuery` method in the example.
+
+```go
+func runSQLQuery(ctx context.Context, db *sql.DB) error {
+    // Create a parent span (Optional)
+    tracer := otel.GetTracerProvider()
+    ctx, span := tracer.Tracer(instrumentationName).Start(ctx, "example")
+    defer span.End()
+
+    runSlowSQLQuery(ctx, db)
+
+    err := query(ctx, db)
+    if err != nil {
+        span.RecordError(err)
+        return err
+    }
+    return nil
+}
+
+func runSlowSQLQuery(ctx context.Context, db *sql.DB) {
+    db.QueryContext(ctx, `SELECT SLEEP(1)`)
+}
+```
+
+This time, we added a new query to the example, which is a slow query that would
+take 1 second to return. Let's see what could happen and how to identify this
+slow query.
+
+To make this change work, we need to rebuild the `client` service.
+
+```sh
+docker compose build client
+docker compose up client
+```
+
+After the client is finished, we can check the trace graph for the trace we just
+generated on Jaeger.
+
+![example of real-world-like Jaeger UI](real-world-like-jaeger-example.png)
+
+From this graph, we know the entire example takes 1 second to complete. The root
+cause of this slowness is not related to the network latency with the database
+and the timestamp query. It is the `SELECT SLEEP(1)` query that leads to the
+slowness.
+
+You can also learn the slowness from the aggregated statistics of the database
+by metrics. And that is the observability the otelsql can provide to know what
+your application is doing with the database.
+
+## Compatibility
+
+You might worry about the compatibility issue with other databases and other
+third-party database frameworks (like ORMs) and wonder how widely this
+instrumentation can be used.
+
+The answer is, from the implementation perspective, as long as the database
+drivers or the database frameworks interact with the database (even not a SQL
+database) through the `database/sql` with context, the `otelsql` should work
+with them.
+
+This is an
+[example](https://github.com/ent/ent/issues/1232#issuecomment-1200405070) that
+shows how otelsql works with Facebook's entity framework for Go.
+
+## Other cool features
+
+### Sqlcommenter support
+
+otelsql integrates [Sqlcommenter](https://google.github.io/sqlcommenter), an
```

Comment:
TIL, that there is a sqlcommenter website, we probably should figure out what to do here since sqlcommenter is part of OTel for a while now (https://github.com/open-telemetry/opentelemetry-sqlcommenter)

Reply:
Do you mean we should create a sqlcommenter doc inside the website, and use an internal link instead of an external link to do the reference?